toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -192,12 +196,48 @@ func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, srpmFile, ou
 		retryDuration = time.Second
 	)
 
-	logBaseName := filepath.Base(srpmFile) + ".log"
+	srpmBase := filepath.Base(srpmFile)
+	logBaseName := srpmBase + ".log"
+	const checkAttempts = 3
+	totalAttempts := buildAttempts
+	if agent.Config().RunCheck && totalAttempts < checkAttempts {
+		totalAttempts = checkAttempts
+	}
 	err = retry.Run(func() (buildErr error) {
 		builtFiles, logFile, buildErr = agent.BuildPackage(srpmFile, logBaseName, outArch, dependencies)
-		return
-	}, buildAttempts, retryDuration)
+		// If the package builds with no errors and RUN_CHECK=y, check logs to see if the %check section passed, and if not, return as the build error.
+		if buildErr != nil {
+			return
+		}
 
+		if agent.Config().RunCheck {
+			file, logErr := os.Open(logFile)
+			// If we can't open the log file, that's a build error.
+			if logErr != nil {
+				logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logfile, logErr)
+				buildErr = logErr				
+				return
+			}
+			defer file.Close()
+			scanner := bufio.NewScanner(file)
+			for scanner.Scan() {
```

Comment:
Ah, sorry about that. Looks like I missed a semicolon when I was providing the example. If you're still interested [**NOT** important, just an FYI]:
```suggestion
			for scanner := bufio.NewScanner(file); scanner.Scan(); {
```

And btw. I got the idea from [here](https://github.com/microsoft/CBL-Mariner/blob/225fbf0b815e0ddc19cc39c43473f5fe44b71303/toolkit/tools/internal/logger/log.go#L139).

Reply:
Ahh, I was curious what was causing the issue with that because it looked fine. Didn't realize the semicolon would fix that :)